components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java

```diff
@@ -765,19 +763,41 @@ protected void sendResponse(HttpServletRequest request, HttpServletResponse resp
 
         // redirect to the caller
         String redirectURL;
-        String commonauthCallerPath = context.getCallerPath();
-
+        String callerPath = context.getCallerPath();
         try {
-            String queryParamsString = "";
-            if (context.getCallerSessionKey() != null) {
-                queryParamsString = FrameworkConstants.SESSION_DATA_KEY + "=" +
-                        URLEncoder.encode(context.getCallerSessionKey(), "UTF-8");
-            }
+            if (!FrameworkUtils.isAbsoluteURI(callerPath)) {
+                ServiceURLBuilder serviceURLBuilder = ServiceURLBuilder.create().addPath(context.getCallerPath());
+                if (context.getCallerSessionKey() != null) {
+                    serviceURLBuilder.addParameter(FrameworkConstants.SESSION_DATA_KEY, URLEncoder.encode(context
+                            .getCallerSessionKey(), "UTF-8"));
+                }
+
+                if (context.isRequestAuthenticated() && context.isRememberMe()) {
+                    serviceURLBuilder.addParameter("chkRemember", "on");
```

Comment:
Shall we move `chkRemember` to a constant as it has been used in multiple places?